packages/react-devtools-extensions/src/utils.js

```diff
@@ -1,10 +1,35 @@
 /* global chrome */
 
-import type {BrowserTheme} from 'react-devtools-shared/src/devtools/views/DevTools';
+import type { BrowserTheme } from 'react-devtools-shared/src/devtools/views/DevTools';
 
-export const IS_EDGE = navigator.userAgent.indexOf('Edg') >= 0;
-export const IS_FIREFOX = navigator.userAgent.indexOf('Firefox') >= 0;
-export const IS_CHROME = IS_EDGE === false && IS_FIREFOX === false;
+let IS_EDGE;
+let IS_FIREFOX;
+let IS_CHROME;
+
+async function getBrowserData() {
+  if (navigator.userAgentData) {
+    const userAgentData = await navigator.userAgentData.getHighEntropyValues([
+      'brands',
+    ]);
```

Comment:
Is `navigator.userAgentData` supported by Firefox? 

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgentData#browser_compatibility

Reply:
For now, we've added a fallback to use navigator.userAgent in browsers that don't support navigator.userAgentData.